chromadb/utils/data_loaders.py

```diff
@@ -16,8 +18,19 @@ def __init__(self, max_workers: int = multiprocessing.cpu_count()) -> None:
                 "The PIL python package is not installed. Please install it with `pip install pillow`"
             )
 
+    #    def _load_image(self, uri: Optional[URI]) -> Optional[Image]:
+    #        return np.array(self._PILImage.open(uri)) if uri is not None else None
+
     def _load_image(self, uri: Optional[URI]) -> Optional[Image]:
-        return np.array(self._PILImage.open(uri)) if uri is not None else None
+        if uri is not None:
+            if uri[0:4] == "http":
+                return np.asarray(
+                    self._PILImage.open(BytesIO(requests.get(uri).content))
```

Comment:
Thanks for taking a look!

Good idea. I'll replace `requests` with `httpx`. 

Coming to auth, [the docs](https://www.python-httpx.org/quickstart/#authentication) are clear enough. I'll add the feature. But then, it will make much more sense to split out this feature (adding images from URLs) into a separate loader class. Like @jeffchuber suggested on the [issues page](https://github.com/chroma-core/chroma/issues/2401)